src/pages/Homepage.tsx

```diff
@@ -1,42 +1,49 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import Carousel from "react-bootstrap/Carousel";
 import CardHome from "../components/CardHome";
 // @ts-ignore
 import workoutImg from "../assests/images/workout.jpg";
 //@ts-ignore
-import healthContentImg from "../assests/images/health-content.jpg";
+// import healthContentImg from "../assests/images/health-content.jpg";
 import styles from "./Homepage.module.css";
+import { CardHomeInterface } from "../interfaces/CardHomeInterface";
 
 const Homepage = () => {
-  //temp data
-  const page = [
-    {
-      id: 1,
-      title: "Workout Plans",
-      description:
-        "Managing and creating weekly workout plans for your convenience",
-      image: workoutImg,
-      link: "workouts",
-    },
-    {
-      id: 2,
-      title: "Health Content",
-      description: "Managing and creating health content for you routine",
-      image: healthContentImg,
-      link: "health-content",
-    },
-  ];
+  const [homeCards, setHomeCards] = useState<CardHomeInterface[]>();
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  const fetchData = async function () {
+    try {
+      const res = await fetch("http://localhost:3000/homepage");
+
+      let error;
+
+      if (!res.ok) {
+        throw new Error(`HTTP error! Status: ${res.status}`);
+        error = res.status;
+      }
+
+      const data = await res.json();
+      setHomeCards(data);
+    } catch (error: any) {
```

Comment:
you could do a little research and look for the correct type to put for this error response.

Reply:
I think you fixed this